services/ansibler/server/loadbalancers.go

```diff
@@ -448,6 +455,51 @@ func setUpNginx(lb *pb.LBcluster, targetedNodepool []*pb.NodePool, directory str
 	return nil
 }
 
+// setUpNodeExporter sets up node-exporter on the LB node
+// return error if not successful, nil otherwise
+func setUpNodeExporter(lb *pb.LBcluster, directory string) error {
+	//create key files for lb nodepools
+	if _, err := os.Stat(directory); os.IsNotExist(err) {
+		if err := os.MkdirAll(directory, os.ModePerm); err != nil {
+			return fmt.Errorf("failed to create directory %s : %w", directory, err)
+		}
+	}
+	if err := utils.CreateKeyFile(lb.ClusterInfo.PrivateKey, directory, fmt.Sprintf("key.%s", privateKeyExt)); err != nil {
```

Comment:
I think that if we are creating key files and whatnot in `setUpNodeExporter()`, we can then remove that from `setUpNginx()`, as files get cleared at the end of `setUpLoadbalancers()`

Reply:
What do you think about creating a separate function for creating ssh keys? IMO it would have sense.

Reply:
I moved it to `setUpLoadbalancers` function.